rts/motoko-rts/src/memory/ic.rs

```diff
@@ -1,14 +1,16 @@
 // This module is only enabled when compiling the RTS for IC or WASI.
 
 use super::Memory;
-use crate::constants::WASM_PAGE_SIZE;
+use crate::constants::{GB, WASM_PAGE_SIZE};
 use crate::gc::incremental::memory_reserve;
-use crate::gc::incremental::partitioned_heap::MAXIMUM_MEMORY_SIZE;
 use crate::rts_trap_with;
 use crate::types::{Bytes, Value, Words};
 use core::arch::wasm64;
 use core::cmp::min;
 
+// TODO: Can we inspect the maximum main memory limit from the IC?
```

Comment:
I don't think we can... even if you could, you might not get that amount on request.

Reply:
(Indeed, if we really want to leave a reliable reserve, perhaps we should actually allocate some extra space beyond ptr unless we are actively using the reserve. But that makes my head hurt)

Reply:
An interesting idea of pre-allocating the reserve. A downside would be that it could be quite wasteful if the reserve is large (or we apply some heuristics of probing that limit when getting close to some "magic" boundaries, e.g. 4GB, 6GB, 8GB, ...).